### PR TITLE
[Feature] `src` support on `script`

### PR DESCRIPTION
**Pull request by stramel**
#34 Initial attempt at adding `src` support to the `script[type="text/markdown"]` tag.

@FredKSchott PTAL and let me know any comments you may have. 😄 I'm sure there will be some code style comments.

### CODE REVIEW

**Comment by stramel** [comment]
@FredKSchott I know you are busy, just a friendly ping :smile:

File: README.md [hunk 1]
```diff
@@ -15,25 +15,27 @@ thing! https://github.com/PolymerLabs/tedium/issues
...
+`<marked-element>` accepts Markdown source and renders it to a child
 element with the class `markdown-html`. This child element can be styled
 as you would a normal DOM element. If you do not provide a child element
 with the `markdown-html` class, the Markdown source will still be rendered,
 but to a shadow DOM child that cannot be styled.
 
-The Markdown source can be specified either via the `markdown` attribute:
+The Markdown source can be specified several ways:
+
+via the `markdown` attribute
```
Comment (FredKSchott) [inline README.md]: For each of these different ways to specify source, turn this into a `###` (cleaning up the wording to be more header-like might be needed as well)

File: demo/example.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+<paper-toolbar>
+  <paper-icon-button icon="menu"></paper-icon-button>
+  <div class="title">Title</div>
+  <paper-icon-button icon="more"></paper-icon-button>
+</paper-toolbar>
+```
+
+_Nifty_ features.
+
+[![Build status](https://travis-ci.org/PolymerElements/marked-element.svg?branch=master)](https://travis-ci.org/PolymerElements/marked-element)
```
Comment (FredKSchott) [inline demo/example.md]: is this to test images? Use of the build status is confusing here, consider using a different image.

File: marked-element.html [hunk 3]
```diff
@@ -149,17 +158,35 @@
         observer: 'render',
         type: Function,
         value: null
+      },
+      /**
+       * A reference to the XMLHttpRequest instance used to generate the
+       * network request.
```
Comment (FredKSchott) [inline marked-element.html:167]: Only set this if a remote request is needed

File: marked-element.html [hunk 4]
```diff
@@ -149,17 +158,35 @@
...
 
     ready: function() {
-      if (!this.markdown) {
-        // Use the Markdown from the first `<script>` descendant whose MIME type starts with
-        // "text/markdown". Script elements beyond the first are ignored.
-        var markdownElement = Polymer.dom(this).querySelector('[type^="text/markdown"]');
-        if (markdownElement != null) {
-          this.markdown = this._unindent(markdownElement.textContent);
-        }
+      if (this.markdown) return;
```
Comment (FredKSchott) [inline marked-element.html]: Formatting: please expand this and others below:
```
if (this.markdown) {
  return;
}
```

File: marked-element.html [hunk 5]
```diff
@@ -149,17 +158,35 @@
...
+
+      // Use the Markdown from the first `<script>` descendant whose MIME type starts with
+      // "text/markdown". Script elements beyond the first are ignored.
+      var markdownElement = Polymer.dom(this).querySelector('[type^="text/markdown"]');
```
Comment (FredKSchott) [inline marked-element.html]: Are there any values other than "text/markdown" that start with "text/markdown"? If so, can you give an example in the comment above?

**Comment by FredKSchott** [comment]
Completely missed this in my inbox overload last week, thanks for the ping. Reviewing now...

File: marked-element.html [hunk 6]
```diff
@@ -149,17 +158,35 @@
...
+      if (!markdownElement) return;
+
+      if (markdownElement.src) {
+        this._request(markdownElement.src);
+      } else {
+        this.markdown = this._unindent(markdownElement.textContent);
```
Comment (FredKSchott) [inline marked-element.html]: Could we support placeholder text while the request is in progress? We could use the script contents as the placeholder, overwritten when the ajax request comes back. 

Something like this:

```
<marked-element>
  <div class="markdown-html"></div>
  <script type="text/markdown" src="../guidelines.md">
    Loading markdown...
  </script>
</marked-element>
```

File: marked-element.html [hunk 7]
```diff
@@ -258,8 +285,26 @@
...
+
+      if (xhr.readyState > 0) return null;
+
+      xhr.addEventListener('loadend', function (e) {
+        var status = this.xhr.status || 0;
+        // Note: if we are using the file:// protocol, the status code will be 0
+        // for all outcomes (successful or otherwise).
+        if (status === 0 || (status >= 200 && status < 300)) {
+          this.markdown = e.target.response;
+        }
```
Comment (FredKSchott) [inline marked-element.html]: Can we support the fail scenario as well? Something as simple as `this.markdown = 'text failed to load';` would add a lot of value.

**Review by FredKSchott — COMMENTED** [review]
🎉 💯  @stramel thanks for tackling this, Overall looks like a good direction design wise. Left a few comments.

File: marked-element.html (hunk 5 above)
Comment (stramel) [reply marked-element.html]: It looks like `text/markdown` is the only thing registered for markdown in the IANA and RFC7763. Would you prefer to use `[type="text/markdown"]` here instead?

File: demo/example.md (hunk 2 above)
Comment (stramel) [reply demo/example.md]: Yeah, was just being lazy 😅 I will find a different image to use, unless you have any suggestions.

File: marked-element.html (hunk 4 above)
Comment (stramel) [reply marked-element.html]: Okay, would you like me to do this for some of the others that exist in here?

File: marked-element.html (hunk 6 above)
Comment (stramel) [reply marked-element.html]: Good idea! Added 👍

File: marked-element.html (hunk 3 above)
Comment (stramel) [reply marked-element.html:167]: Done

File: marked-element.html (hunk 7 above)
Comment (stramel) [reply marked-element.html]: Added a error event as well. Should we support both? I think maybe firing the event and allowing the consumer to do what they want with the content.

File: README.md (hunk 1 above)
Comment (stramel) [reply README.md]: Attempted this, definitely open to suggestions

File: demo/example.md (hunk 2 above)
Comment (stramel) [reply demo/example.md]: Used a polymer.svg harvested from `iron-image`

File: demo/example.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+<paper-toolbar>
+  <paper-icon-button icon="menu"></paper-icon-button>
+  <div class="title">Title</div>
+  <paper-icon-button icon="more"></paper-icon-button>
+</paper-toolbar>
+```
+
+_Nifty_ features.
+
+![Polymer](polymer.svg)
```
Comment (stramel) [inline demo/example.md]: Could optionally just remove the image from the demo. The `polymer.svg` is 12.1kb

File: marked-element.html [hunk 9]
```diff
@@ -149,17 +161,36 @@
...
+      // Use the Markdown from the first `<script>` descendant whose MIME type starts with
+      // "text/markdown". Script elements beyond the first are ignored.
+      var markdownElement = Polymer.dom(this).querySelector('[type="text/markdown"]');
+      if (!markdownElement) {
+        return;
+      }
+
+      this.markdown = this._unindent(markdownElement.textContent);
+
+      if (markdownElement.src) {
```
Comment (stramel) [inline marked-element.html:192]: Remote src will now just override the default content.

**Comment by stramel** [comment]
@FredKSchott Updated based on comments

**Comment by stramel** [comment]
@FredKSchott Friendly ping! Take a look at the fallback and error event. Thanks!

**Comment by stramel** [comment]
@e111077 do you have any thoughts?

File: demo/example.md (hunk 8 above)
Comment (e111077) [reply demo/example.md]: Can you please? I don't think the svg is that necessary.

**Review by e111077 — CHANGES_REQUESTED** [review]

**Comment by e111077** [comment]
other than that LGTM

File: demo/example.md (hunk 8 above)
Comment (stramel) [reply demo/example.md]: Sure thing!

**Comment by stramel** [comment]
🎉 Thanks @e111077!

**Comment by FredKSchott** [comment]
Thanks for 💪 on this @stramel!

**Comment by stramel** [comment]
You're welcome! Glad to help out!

**Pull request merged**